### ISSUE TITLE
Switch from YCM to NCM

### ISSUE THREAD

**Issue by ZeroKnight** [ZeroKnight/dotfiles#17]
It would probably be a good thing to switch to [NCM](https://github.com/roxma/nvim-completion-manager); it has quite a few neat features that YCM lacks (though the reverse is also true for a few things).

A few highlights (see Evernote for deeper analysis):

- Runs completion sources in parallel
- Scoping, basically "smart completion"; allow multiple filetype completion sources at once (e.g. CSS completion in an HTML <style> tag)
- Has a LOT of sources available, including ones from deoplete, and creating sources seems to be less involved than YCM

Overall, NCM is lighter, less complex and likely to be faster than YCM. Also, **no compiled component to worry about when updating**.

**Comment by ZeroKnight** [comment]
On hold due to unacceptable performance from the neoinclude completer

**Comment by ZeroKnight** [comment]
NCM is fully set up for now on a separate branch. Will see if it's better overall in the long run.
